auth/client/iam/openid4vp.go

```diff
@@ -123,10 +124,14 @@ func (c *OpenID4VPClient) PresentationDefinition(ctx context.Context, endpoint s
 	return presentationDefinition, nil
 }
 
-func (c *OpenID4VPClient) AuthorizationServerMetadata(ctx context.Context, webdid did.DID) (*oauth.AuthorizationServerMetadata, error) {
+func (c *OpenID4VPClient) AuthorizationServerMetadata(ctx context.Context, oauthIssuer string) (*oauth.AuthorizationServerMetadata, error) {
```

Comment:
This is a naming conflict on the `issuer` term we keep running into.
The oauth `/.well-known/authorization-server` metadata has an `issuer` field that is the identity of the AS. This `issuer` is what the metadata URL is derived from, and corresponds to did:web identity in URL form. So `oauthIssuer` is just a more generic name and not related to the issuer-holder-verifier model. (hence prefix `oauth`)

We need this change because OpenID4VCI defines a `/.well-known/openid-credential-issuer` metadata that contains field `authorization_servers` that lists multiple URLs, not DIDs.

Reply:
would you prefer 2 methods for convenience?

Reply:
aha, then just explain this on the interfaces